libs/wingc/src/type_check.rs

```diff
@@ -2045,6 +2062,55 @@ impl<'a> TypeChecker<'a> {
 				self.inner_scopes.push(statements);
 			}
 			StmtKind::Break | StmtKind::Continue => {}
+			StmtKind::IfLet {
+				condition,
+				statements,
+				var_name,
+				else_statements,
+			} => {
+				let cond_type = self.type_check_exp(condition, env);
+				self.validate_type_is_optional(cond_type, condition);
+
+				// Technically we only allow if let statements to be used with optionals
+				// and above validate_type_is_optional method will attach a diagnostic error if it is not.
+				// However to avoid panics this will allow code to continue if the type is not an optional
+				// and complete the type checking process for additional errors.
+				let var_type = if cond_type.is_option() {
+					cond_type
+						.optional_type()
+						.expect("Optional types should always have a type")
+				} else {
+					cond_type
+				};
```

Comment:
I think we can avoid making an unsafe `expect` / `unwrap` call here
```suggestion
				let var_type = match cond_type.optional_type() {
					Some(t) => t,
					None => cond_type,
				};
```

Reply:
Check out @MarkMcCulloh's neat helper:
```suggestion
				let var_type = if cond_type.maybe_unwrap_optional();
```
We can also delete the `optional_type()` method above as I think we don't really need it.